embedding/embedlite/embedshared/nsWindow.cpp

```diff
@@ -276,11 +285,57 @@ layers::LayersId nsWindow::GetRootLayerId() const
   return mCompositorSession ? mCompositorSession->RootLayerTreeId() : layers::LayersId{0};
 }
 
-void nsWindow::SetContentController(mozilla::layers::GeckoContentController *aController)
+#ifdef MOZ_LOGGING
+static void LogControllerStack(std::list<EmbedContentController *> &controllers)
+{
+  int pos = 0;
+  for (std::list<EmbedContentController *>::const_iterator it = controllers.begin(); it != controllers.end(); ++it) {
+    LOGT("Stack controller %d : %u", pos, (*it)->GetUniqueID());
+    ++pos;
+  }
+}
+#endif // MOZ_LOGGING
+
+void nsWindow::SetContentController(EmbedContentController *aController)
```

Comment:
How does nsWindow::Reserve(EmbedContentController *aController) or nsWindow::Activate(EmbedContentController *aController) sound like?

Maybe this could Reserve (SetContentController) & Release (DropContentController) or Activate (SetContentController) & Deactivate (DropContentController) pair. Now that I said Activate & Deactivate that pair sounds more natural to me. What do you think?